source/common/ssl/context_config_impl.cc

```diff
@@ -77,7 +78,42 @@ ClientContextConfigImpl::ClientContextConfigImpl(const Json::Object& config)
 ServerContextConfigImpl::ServerContextConfigImpl(const envoy::api::v2::DownstreamTlsContext& config)
     : ContextConfigImpl(config.common_tls_context()),
       require_client_certificate_(
-          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, require_client_certificate, false)) {
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, require_client_certificate, false)),
+      session_ticket_keys_([&config] {
+        std::vector<std::vector<uint8_t>> ret;
+
+        switch (config.session_ticket_keys_type_case()) {
+        case envoy::api::v2::DownstreamTlsContext::kSessionTicketKeys:
+          for (const auto& datasource : config.session_ticket_keys().keys()) {
+            switch (datasource.specifier_case()) {
+            case envoy::api::v2::DataSource::kFilename: {
+              const std::string key_data = Filesystem::fileReadToEnd(datasource.filename());
+              ret.push_back(std::vector<uint8_t>(key_data.begin(), key_data.end()));
+              break;
+            }
+            case envoy::api::v2::DataSource::kInline: {
+              const std::string& key_data = datasource.inline_();
+              ret.push_back(std::vector<uint8_t>(key_data.begin(), key_data.end()));
+              break;
+            }
+            default:
+              throw EnvoyException(fmt::format("Unexpected DataSource::specifier_case(): {}",
+                                               datasource.specifier_case()));
+            }
+          }
+          break;
+        case envoy::api::v2::DownstreamTlsContext::kSessionTicketKeysSdsSecretConfig:
+          NOT_IMPLEMENTED;
+          break;
+        case envoy::api::v2::DownstreamTlsContext::SESSION_TICKET_KEYS_TYPE_NOT_SET:
```

Comment:
Right now the default case throws, because it got a format it doesn't understand.

This case will probably be the most-common case: no keys set at all, which should not be an error.  Even if we don't think the default case should throw, it should at least log or something, whereas this case should be silent I think.

Reply:
Ah, I see. Thanks!